### PR TITLE
Documenting SignerSignEx3

### PR DESCRIPTION
**Pull request by rotruxal**
Added 8 new files in total. This PR will likely need some formatting tweaks. A few outstanding questions:
- What is ms.assetid? How do I determine that?
- What is the correct way to specify links to docs within the folder? `./`? to docs outside the folder? `https://...`?

### CODE REVIEW

**Comment by alvinashcraft** [comment]
Hi @rotruxal. Thanks for your contribution. I'll take a look at the files soon. For now, here are some answers to your questions:

1. The ms.assetid is just a unique GUID. You can generate them in many places, like [this site](https://guidgenerator.com/).
2. If a file is in the same folder, you can just use the file name `[this file](thisfile.md)`. If it's in another folder, use the URL path without the learn.microsoft.com and region at the front. For example, `https://learn.microsoft.com/en-us/windows/apps/` would be just `/windows/apps/` or `[Windows dev home](/windows/apps/)`.

Does this help?

**Comment by alvinashcraft** [comment]
@rotruxal - The other thing you will need to do is edit the toc.yml file in the SecCrypto folder so these new files appear in the left-hand navigation tree in the docs.

File: desktop-src/SecCrypto/signersignex3.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+> [!Note]  
+> This function has no associated header file or import library. To call this function, you must create a user-defined header file and use the [**LoadLibrary**](/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibrarya) and [**GetProcAddress**](/windows/win32/api/libloaderapi/nf-libloaderapi-getprocaddress) functions to dynamically link to Mssign32.dll.
+
+ 
+
+## Syntax
+
+
+```C++
+HRESULT WINAPI SignerSignEx2(
```
Comment (clairernovotny) [inline desktop-src/SecCrypto/signersignex3.md]:
```suggestion
HRESULT WINAPI SignerSignEx3(
```

File: desktop-src/SecCrypto/signersignex3.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+- SignerSignEx3
+api_type: 
+- DllExport
+api_location: 
+- Mssign32.dll
+---
+
+# SignerSignEx2 function
+
+The **SignerSignEx2** function signs and time stamps the specified file, allowing multiple nested signatures.
```
Comment (clairernovotny) [inline desktop-src/SecCrypto/signersignex3.md]:
```suggestion
The **SignerSignEx3** function signs and time stamps the specified file, allowing multiple nested signatures.
```

File: desktop-src/SecCrypto/signersignex3.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+- kbSyntax
+api_name: 
+- SignerSignEx3
+api_type: 
+- DllExport
+api_location: 
+- Mssign32.dll
+---
+
+# SignerSignEx2 function
```
Comment (clairernovotny) [inline desktop-src/SecCrypto/signersignex3.md]:
```suggestion
# SignerSignEx3 function
```

**Review by clairernovotny — CHANGES_REQUESTED** [review]

**Comment by rotruxal** [comment]
I added asset ids to all the files and more links for existing structures.

**Comment by rotruxal** [comment]
Ok does this look ready for publishing?

**Comment by alvinashcraft** [comment]
I'll take a look at the latest files now. Thanks!

File: desktop-src/SecCrypto/pfn-authenticode-digest-sign-ex-withfilehandle.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+</dt> <dd>
+
+Optional, receives the cert chain of the signer cert.
+
+</dd></dl>
+
+## Return value
+
+If the function succeeds, the function returns S_OK.
+If the function fails, it returns an HRESULT value that indicates the error. For a list of common error codes, see [Common HRESULT Values](https://learn.microsoft.com/en-us/windows/win32/seccrypto/common-hresult-values).
```
Comment (alvinashcraft) [inline desktop-src/SecCrypto/pfn-authenticode-digest-sign-ex-withfilehandle.md]:
```suggestion
If the function fails, it returns an HRESULT value that indicates the error. For a list of common error codes, see [Common HRESULT Values](/windows/win32/seccrypto/common-hresult-values).
```

File: desktop-src/SecCrypto/pfn-authenticode-digest-sign-ex.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+</dt> <dd>
+
+Optional, receives the cert chain of the signer cert.  
+
+</dd> </dl>
+
+## Return value
+
+If the function succeeds, the function returns S_OK.
+If the function fails, it returns an HRESULT value that indicates the error. For a list of common error codes, see [Common HRESULT Values](https://learn.microsoft.com/en-us/windows/win32/seccrypto/common-hresult-values).
```
Comment (alvinashcraft) [inline desktop-src/SecCrypto/pfn-authenticode-digest-sign-ex.md]:
```suggestion
If the function fails, it returns an HRESULT value that indicates the error. For a list of common error codes, see [Common HRESULT Values](/windows/win32/seccrypto/common-hresult-values).
```

File: desktop-src/SecCrypto/pfn-authenticode-digest-sign-withfilehandle.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+</dt> <dd>
+
+Pointer to [**CRYPT_DATA_BLOB**](/windows/win32/api/wincrypt/ns-wincrypt-crypt_integer_blob) which receives the signed digest.
+
+</dd> </dl>
+
+## Return value
+
+If the function succeeds, the function returns S_OK.
+If the function fails, it returns an HRESULT value that indicates the error. For a list of common error codes, see [Common HRESULT Values](https://learn.microsoft.com/en-us/windows/win32/seccrypto/common-hresult-values).
```
Comment (alvinashcraft) [inline desktop-src/SecCrypto/pfn-authenticode-digest-sign-withfilehandle.md]:
```suggestion
If the function fails, it returns an HRESULT value that indicates the error. For a list of common error codes, see [Common HRESULT Values](/windows/win32/seccrypto/common-hresult-values).
```

File: desktop-src/SecCrypto/pfn-authenticode-digest-sign.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+</dt> <dd>
+
+Pointer to [**CRYPT_DATA_BLOB**](/windows/win32/api/wincrypt/ns-wincrypt-crypt_integer_blob) which receives the signed digest.
+
+</dd> </dl>
+
+## Return value
+
+If the function succeeds, the function returns S_OK.
+If the function fails, it returns an HRESULT value that indicates the error. For a list of common error codes, see [Common HRESULT Values](https://learn.microsoft.com/en-us/windows/win32/seccrypto/common-hresult-values).
```
Comment (alvinashcraft) [inline desktop-src/SecCrypto/pfn-authenticode-digest-sign.md]:
```suggestion
If the function fails, it returns an HRESULT value that indicates the error. For a list of common error codes, see [Common HRESULT Values](/windows/win32/seccrypto/common-hresult-values).
```

File: desktop-src/SecCrypto/pfn-authenticode-digest-sign.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
+
+## Remarks  
+
+The parameter pSignedDigest->pbData must be allocated by calling HeapAlloc passing the result of GetProcessHeap() as the first parameter.
```
Comment (alvinashcraft) [inline desktop-src/SecCrypto/pfn-authenticode-digest-sign.md]: So, I would add more code formatting throughout these files, but we can always come back with another PR later to address that. Here's an example:
```suggestion
The parameter `pSignedDigest->pbData` must be allocated by calling `HeapAlloc` passing the result of `GetProcessHeap()` as the first parameter. 
```

File: desktop-src/SecCrypto/signer-digest-sign-info.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+
+
+ 
+
+</dd> <dt>
+
+**pfnAuthenticodeDigestSign**
+</dt> <dd>
+
+Pointer to the [**PFN_AUTHENTICODE_DIGEST_SIGN**](./pfn-authenticode-digest-sign.md) callback function. Required if dwDigestSignChoice==DIGEST_SIGN.
```
Comment (alvinashcraft) [inline desktop-src/SecCrypto/signer-digest-sign-info.md]:
```suggestion
Pointer to the [**PFN_AUTHENTICODE_DIGEST_SIGN**](pfn-authenticode-digest-sign.md) callback function. Required if dwDigestSignChoice==DIGEST_SIGN.
```

**Review by alvinashcraft — COMMENTED** [review]
Looks good overall. I left a few suggestions inline (mostly updating links).

**Comment by rotruxal** [comment]
ok I've incorporated all the feedback. How does it look now?

**Review by alvinashcraft — APPROVED** [review]
I saw three suggestions that didn't get committed yet (link changes), so I committed them. Everything else looks good to me. Thanks! Does @clairernovotny need to take another look before we merge?

**Pull request merged**